Leanplum-SDK/Classes/LPDatabase.m

```diff
@@ -0,0 +1,226 @@
+//
+//  LPDatabase.m
+//  Leanplum
+//
+//  Created by Alexis Oyama on 6/9/17.
+//  Copyright (c) 2017 Leanplum, Inc. All rights reserved.
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+
+#import "LPDatabase.h"
+#import "LPFileManager.h"
+#import "LeanplumInternal.h"
+#import "Constants.h"
+#import <sqlite3.h>
+
+static sqlite3 *sqlite;
+static BOOL retryOnCorrupt;
+static BOOL willSendErrorLog;
+
+@implementation LPDatabase
+
+- (id)init
+{
+    if (self = [super init]) {
```

Comment:
Gotya, maybe rename to sharedDatabase or sharedInstance as I believe this is the common standard: https://stackoverflow.com/a/7569010/2302437